chain/client/src/client_actor.rs

```diff
@@ -179,8 +185,27 @@ impl ClientActor {
             chunk_request_retry_next_attempt: now,
             sync_started: false,
             state_parts_task_scheduler: Box::new(move |msg: ApplyStatePartsRequest| {
-                if let Err(_) = state_parts_actor_addr.try_send(msg) {
-                    panic!("Can't send message to StatePartsActor");
+                if let Err(err) = sync_jobs_actor_addr.try_send(msg) {
+                    match err {
+                        SendError::Full(request) => {
+                            sync_jobs_actor_addr.do_send(request);
+                        }
+                        SendError::Closed(_) => {
+                            panic!("Can't send message to SyncJobsActor, mailbox is closed");
+                        }
+                    }
+                }
+            }),
+            block_catch_up_scheduler: Box::new(move |msg: BlockCatchUpRequest| {
+                if let Err(err) = sync_jobs_actor_addr2.try_send(msg) {
+                    match err {
+                        SendError::Full(request) => {
+                            sync_jobs_actor_addr2.do_send(request);
+                        }
+                        SendError::Closed(_) => {
+                            panic!("Can't send message to SyncJobsActor, mailbox is closed");
+                        }
```

Comment:
this code could probably be refactored to avoid duplication?

Reply:
Agreed, especially we will add another type in the future for state split jobs.